### PR TITLE
PHOENIX-6339 Older client using aggregate queries shows incorrect res…

### PR DESCRIPTION
**Pull request by kadirozde**
…ults

### CODE REVIEW

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 32s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ 4.x Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  14m 42s |  4.x passed  |
| +1 :green_heart: |  compile  |   1m  3s |  4.x passed  |
| +1 :green_heart: |  checkstyle  |   1m 25s |  4.x passed  |
| +1 :green_heart: |  javadoc  |   0m 45s |  4.x passed  |
| +0 :ok: |  spotbugs  |   3m  0s |  phoenix-core in 4.x has 945 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 56s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  4s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  4s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m 30s |  phoenix-core: The patch generated 40 new + 2377 unchanged - 29 fixed = 2417 total (was 2406)  |
| +1 :green_heart: |  whitespace  |   0m  1s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 43s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   3m 14s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 128m 43s |  phoenix-core in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 14s |  The patch does not generate ASF License warnings.  |
|  |   | 163m 51s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1111/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1111 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile |
| uname | Linux e3727bcf4df8 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | 4.x / 51715b3 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1111/1/artifact/yetus-general-check/output/diff-checkstyle-phoenix-core.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1111/1/testReport/ |
| Max. process+thread count | 5655 (vs. ulimit of 30000) |
| modules | C: phoenix-core U: phoenix-core |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1111/1/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: phoenix-core/src/main/java/org/apache/phoenix/util/ScanUtil.java [hunk 1]
```diff
@@ -1304,21 +1304,20 @@ public static PagedFilter getPhoenixPagedFilter(Scan scan) {
...
             pageSizeMs = Bytes.toLong(pageSizeMsBytes);
+            pageSizeMs = (long) (pageSizeMs * factor);
         }
         return pageSizeMs;
     }
 
-    public static long getPageSizeMsForRegionScanner(Scan scan) {
-        return (long) (getPageSizeMs(scan) * 0.6);
-    }
+    public static long getPageSizeMsForRegionScanner(Scan scan)  { return getPageSizeMs(scan, 0.6); }
```
Comment (jpisaac) [inline phoenix-core/src/main/java/org/apache/phoenix/util/ScanUtil.java:1317]: What is/should be the thought process in choosing the factor? May be good to add some comments around it?

**Review by jpisaac — COMMENTED** [review]
Some request for comments, otherwise LGTM.

File: phoenix-core/src/main/java/org/apache/phoenix/util/ScanUtil.java (hunk 1 above)
Comment (kadirozde) [reply phoenix-core/src/main/java/org/apache/phoenix/util/ScanUtil.java:1317]: The comments are just right there before getPageSizeMs()

**Review by jpisaac — APPROVED** [review]

**Review by yanxinyi — APPROVED** [review]
downloaded the patch and built/tested it locally against the 4.15 client. 
LGTM +1

**Review by gjacoby126 — APPROVED** [review]
+1, thanks @kadirozde

**Review by abhishek-chouhan — APPROVED** [review]

**Review by ChinmaySKulkarni — APPROVED** [review]
+1, thanks @kadirozde

**Pull request merged**